Various python

teh opened the pull request:
###### Motivation for this change

Don't merge just yet, running nox-review fails locally so I want to try the `pr` option.

```
error: while querying the derivation named ‘tex4ht-1.0.2009_06_11_1038’:
while evaluating the attribute ‘buildPhase’ of the derivation ‘tex4ht-1.0.2009_06_11_1038’ at /home/tom/.nox/nixpkgs/pkgs/tools/typesetting/tex/tex4ht/default.nix:4:3:
cannot coerce null to a string, at /home/tom/.nox/nixpkgs/pkgs/tools/typesetting/tex/tex4ht/default.nix:15:16
```

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

teh committed:
Logbook: 0.11.3 -> 1.0.0

teh committed:
pytables: 3.2.2 -> 3.3.0

teh commented:
Still no luck, not sure what's up, could be related to 2aff29c7b202725933dd5306eed4241fe86be0b9 by @vcunat ?

In any case, I build by hand:

```
nix-build -A pythonPackages.tables
nix-build -A pythonPackages.Logbook
nix-build -A python3Packages.tables
nix-build -A python3Packages.Logbook
```

and everything succeeded

vcunat commented:
Ah, yes.  It should be OK now.  Some things don't behave very smart, unfortunately.

teh committed:
vega: add pytest as a build-input.

It's unclear how this worked before but it clearly did because we
have vega running in production.

teh committed:
altair: 1.0.0 -> 1.2.0

Also disabling tests because the test runner picks up too much and
then fails.

teh commented:
I get some bizarre errors for rpy2 when building on debian [1] but on nixos `nox-review pr 22873`  now succeeds after fixing unrelated failures in vega and altair.

Can't wait for PR testing by Hydra!

[1] - linking again the local readline .. bizarre, but rpy2 does a lot of weird stuff anyway due to R being weird.
```
ldd /nix/store/dsg3kdf3z1gc4ycdk74b11s51svywix9-python2.7-rpy2-2.8.2/lib/python2.7/site-packages/rpy2/rinterface/_rinterface.so  | grep /lib/x
	libreadline.so.6 => /lib/x86_64-linux-gnu/libreadline.so.6 (0x00007f21a820f000)
	libtinfo.so.5 => /lib/x86_64-linux-gnu/libtinfo.so.5 (0x00007f21a37cc000)
```

vcunat commented:
You would avoid most of these problems by using nix's sandboxing &ndash; that way the builders won't see any `/usr/` or `/lib/`.

teh commented:
@vcunat - thanks. My laptop runs a single-user nix packages on debian which is a slightly unusual arrangement. I'm not sure that's worth blocking on.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -417,15 +417,16 @@ in {
-
+    # Tests try to execute notebooks that aren't tests:
+    # TypeError: calling <function tutorial at 0x7fffe1bb8048> returned None, not a test
+    doCheck = false;
```
maybe the test runner should be invoked differently?

teh replied:
figured it out, needs a UTF locale as well, hope this translates to hydra (works locally).

FRidh commented:
@teh Could you rebase and update this PR?

teh commented:
rebased, PTAL

vcunat commented:
I see `python3Packages.altair` fails tests, but it was so before this PR as well...
